odoo/addons/base/models/ir_property.py

```diff
@@ -341,49 +344,61 @@ def clean(value):
     def search_multi(self, name, model, operator, value):
         """ Return a domain for the records that match the given condition. """
         default_matches = False
-        include_zero = False
+        negate = False
+
+        # For "is set" and "is not set", same logic for all types
+        if operator == 'in' and False in value:
+            operator = 'not in'
+            negate = True
+        elif operator == 'not in' and False not in value:
+            operator = 'in'
+            negate = True
+        elif operator in ('!=', 'not like', 'not ilike') and value:
+            operator = TERM_OPERATORS_NEGATION[operator]
+            negate = True
+        elif operator == '=' and not value:
+            operator = '!='
+            negate = True
 
         field = self.env[model]._fields[name]
+
         if field.type == 'many2one':
-            comodel = field.comodel_name
             def makeref(value):
-                return value and '%s,%s' % (comodel, value)
-            if operator == "=":
-                value = makeref(value)
-                # if searching properties not set, search those not in those set
-                if value is False:
-                    default_matches = True
```

Comment:
Oh, we actually changed that behavior, which was introduced by dd88f3c82f8be67fc2224e7107a5511675873718.
The condition `(fname, '=', False)` was apparently a hack to retrieve records with no explicit value.
But the test that goes with it does not consider the case with a default value...
I wonder whether we actually use this hack, which probably very few people know about.

Reply:
Well, according to the commit message, `(fname, '=', False)` should return the records for which the value is not set.  So, if your field is a many2one and the default value is not `False`, the condition should return the records that have the default value, which is not `False`.  That really sucks.  I hope nobody relies on that.

Reply:
I don't understand in the same way. IMO this fix an issue because the default value at this time was always False.
"When searching if a many2one property field is not set, there may be less
results since only the ones with a reference set to NULL are returned." -> (fname, '=', False) doesn't return anything when the ir_property doesn't exist (not force to False).

But yes the behavior has sightly changed, but it is legit, and it is unlikely that someone relies on a buggy search (with 65 issues).

Reply:
> IMO this fix an issue because the default value at this time was always False.

I wouldn't assume that.  For instance, `property_account_payable_id` on partners is company-dependent and even _required_ (which only impacts the form view).  The declaration of that field implies that there should be a truthy default value for that field.  And there is.

> it is unlikely that someone relies on a buggy search

I wouldn't bet on that.  I have seen things like that.  What people can do is beyond human understanding :smile:

Reply:
Yes you are right.
Then what can we do? 

- Target only the 15.0 (Ok for me, the version is new then if there is an issue it will be in our codebase) ? 
- Remove only that part is not a good idea (and maybe someone relies on our other bugs)
- Or don't change the target. There is a possibility that we break few custom flows, but there are more chance that we fix some features (like custom search).

Reply:
I would go for the current proposal: remove that hack.
I don't see a reason to search for records that have an _implicit_ value, which is basically an implementation detail.
But we need more opinions on this question.